Description: Fix the init script

--- a/dyfi-update
+++ b/dyfi-update
@@ -1,25 +1,33 @@
 #!/bin/sh
-#
-#	dyfi-update SysV init script for starting and stopping
-#	dyfi-update.pl - A perl client for updating dy.fi
-#	hostnames automatically
-#
-#	by dy.fi admins, admin at dy dot fi
-#
+### BEGIN INIT INFO
+# Provides:          dyfi
+# Required-Start:    $remote_fs $syslog
+# Required-Stop:     $remote_fs $syslog
+# Default-Start:     2 3 4 5
+# Default-Stop:
+# Short-Description: dyfi-update SysV init script
+# Description:       SysV init script for starting and stopping
+#                    dyfi-update.pl - A perl client for updating dy.fi
+#                    hostnames automatically.
+### END INIT INFO
 
+set -e
 # Specify the location of the pid file here
 # (this must be the same as in dyfi-update.conf!)
 PIDFILE=/var/run/dyfi-update.pid
 # Specify the location of the configuration file:
-CFGFILE=/usr/local/etc/dyfi-update.conf
+CFGFILE=/etc/dyfi-update.conf
 
-PATH=/sbin:/usr/sbin:/usr/local/sbin:/bin:/usr/bin:/usr/local/bin
+DAEMON=/usr/share/dyfi/dyfi-update.pl
+
+# Get lsb functions
+. /lib/lsb/init-functions
 
 # See how we were called.
 case "$1" in
   start)
 	echo "Starting dy.fi updating daemon..."
-	dyfi-update.pl -f $CFGFILE &
+	start-stop-daemon --start --background --make-pidfile --pidfile $PIDFILE --exec $DAEMON -- -f $CFGFILE
 	[ $? -eq 0 ] && [ -d /var/lock/subsys ] && touch /var/lock/subsys/dyfi-update
         ;;
   stop)
@@ -35,8 +43,25 @@
 	fi
   	
 	;;
+  restart)
+	$0 stop
+	sleep 1
+	$0 start
+	;;
+  reload|force-reload)
+	log_begin_msg "Reloading dy.fi updating daemon..."
+	restart
+	log_end_msg $?
+	;;
+  status)
+	if [ -f $PIDFILE ]; then
+		echo "dy.fi is running"
+	else
+		echo "dy.fi is not running (no pid file)"
+	fi
+	;;
   *)
-        echo "Usage: dyfi-update {start|stop}"
+        echo "Usage: dyfi-update {start|stop|restart|reload|force-reload|status}"
         exit 1
 esac
 
