Description: Remove embedded encode_base64 function

--- dyfi-1.2.0.orig/dyfi-update.pl
+++ dyfi-1.2.0/dyfi-update.pl
@@ -92,6 +92,7 @@ require 5.002;
 use Socket;
 use strict;
 use Fcntl qw(:flock);
+use MIME::Base64;
 
 my ($debug, $log, $pidfile, $cfgfile,
 	$update_host, $update_uri, $update_port,
@@ -258,23 +259,6 @@ sub pipehandler {
 }
 
 #
-#	Base64 encoding code from MIME::Base64
-#	which is Copyright 1995-1999, 2001 Gisle Aas.
-#
-sub encode_base64 ($) {
-	my $res = "";
-	pos($_[0]) = 0;                          # ensure start at the beginning
-	
-	$res = join '', map( pack('u',$_)=~ /^.(\S*)/, ($_[0]=~/(.{1,45})/gs));
-	
-	$res =~ tr|` -_|AA-Za-z0-9+/|;               # # help emacs
-	# fix padding at the end
-	my $padding = (3 - length($_[0]) % 3) % 3;
-	$res =~ s/.{$padding}$/'=' x $padding/e if $padding;
-	return $res;
-}
-
-#
 #	cleanup whitespace
 #
 
